src/tools/Repl/Program.cs

```diff
@@ -138,11 +164,23 @@ public static void REPL(TextReader input, bool echo)
                     // variable assignment: Set( <ident>, <expr> )
                     if (TryMatchSet(expr, out var varName, out var varValue))
                     {
-                        Console.WriteLine(varName + ": " + PrintResult(varValue));
+                        Console.WriteLine(varName + ": " + PrintResult(varValue));
+                        output?.WriteLine(varName + ": " + PrintResult(varValue));
+                    }
+
+                    // IR pretty printer: IR( <expr> )
+                    else if ((match = Regex.Match(expr, @"^\s*IR\((?<expr>.*)\)\s*$", RegexOptions.Singleline)).Success)
+                    {
+                        var opts = new ParserOptions() { AllowsSideEffects = true };
+                        var cr = _engine.Check(match.Groups["expr"].Value, options: opts);
+                        var ir = cr.PrintIR();
+                        Console.WriteLine(ir);
+                        output?.WriteLine(ir);
                     }
 
                     // formula definition: <ident> = <formula>
-                    else if ((match = Regex.Match(expr, @"^\s*(?<ident>\w+)\s*=(?<formula>.*)$", RegexOptions.Singleline)).Success)
+                    else if ((match = Regex.Match(expr, @"^\s*(?<ident>[a-zA-Z]\w+)\s*=(?<formula>.*)$", RegexOptions.Singleline)).Success &&

```

Comment:
>a-zA-Z [](http://example.com/codeflow?start=72&length=6)

numbers, underscore ?

Reply:
Good catch.  There are a few bugs here:
* It should support underscore, but not numbers.  This is why I had changed it to begin with.  Numbers are actually supported with single quotes.
* Single quotes aren't supported
* As currently written, it would require at least two characters, when one should suffice.

Updates made.